traffic_portal/app/src/modules/private/configure/deliveryServices/urlSigKeys/DeliveryServiceUrlSigKeysController.js

```diff
@@ -0,0 +1,100 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var DeliveryServiceUrlSigKeysController = function(deliveryService, urlSigKeys, $scope, $state, locationUtils, deliveryServiceService, deliveryServiceUrlSigKeysService, $uibModal) {
+	$scope.deliveryService = deliveryService;
+	$scope.urlSigKeys = Object.keys(urlSigKeys).map(function(key) {
+			return {sortBy: parseInt(key.slice(3)), label: key, urlSigKey: urlSigKeys[key]};
+	});
+
+	$scope.generateUrlSigKeys = function() {
+        var params = {
+            title: 'Confirmation required',
+            message: 'Are you sure you want to generate new URL signature keys for ' + deliveryService.displayName + '?'
+        };
+        var modalInstance = $uibModal.open({
+            templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+            controller: 'DialogConfirmController',
+            size: 'md',
+            resolve: {
+                params: function () {
+                    return params;
+                }
+            }
+        });
+        modalInstance.result
+            .then(
+            function() {
+                deliveryServiceUrlSigKeysService.generateUrlSigKeys(deliveryService.xmlId).then(
+                function() {
+                    $scope.refresh();
+                });
+            });
+	};
+
+	$scope.refresh = function() {
+		$state.reload(); // reloads all the resolves for the view
+	};
+
+	$scope.selectCopyFromDS = function() {
+        var params = {
+            title: 'Copy URL Sig Keys to: ' + deliveryService.displayName,
+            message: "Please select a Delivery Service to copy from:",
+            label: "xmlId"
+        };
+        var modalInstance = $uibModal.open({
+            templateUrl: 'common/modules/dialog/select/dialog.select.tpl.html',
+            controller: 'DialogSelectController',
+            size: 'md',
+            resolve: {
+                params: function () {
+                    return params;
+                },
+                collection: function(deliveryServiceService) {
+                    return deliveryServiceService.getDeliveryServices({ signed: true })
+                    .then(function(result){
+                    	return _.filter(result, function(ds){
+                    		return ds.id !== deliveryService.id;
+                    	})
+                    });
+                }
+            }
+        });
+        modalInstance.result.then(function(copyFromDs) {
+            deliveryServiceUrlSigKeysService.copyUrlSigKeys(deliveryService.xmlId, copyFromDs.xmlId)
+            .then(
+           		function() {
+            	$state.reload();
+        	});
+        });
+    };
+
+	$scope.navigateToPath = locationUtils.navigateToPath;
+
+	angular.element(document).ready(function () {
+		$('#regexesTable').dataTable({
```

Comment:
this id doesn't seem quite right. regexesTable? i know it's arbitrary but you might want to fix that